connect_ferc1_to_eia.py

```diff
@@ -16,101 +47,50 @@
-               replace(plant_part_rename))
-
-    train_df_ids = compiler_mul.assign_record_id_eia(test_df)
-    train_df_ids['plant_part_og'] = train_df_ids['plant_part']
-    train_df_ids = (
-        train_df_ids.set_index(['record_id_eia']).
-        merge(compiler_mul.plant_parts_df[['true_gran', 'appro_part_label',
-                                           'appro_record_id_eia']],
-              how='left', right_index=True, left_index=True).
-        assign(plant_part=lambda x: x['appro_part_label'],
-               record_id_eia=lambda x: x['appro_record_id_eia']).
-        reset_index(drop=True))
-    # train_df_ids = compiler_mul.assign_record_id_eia(train_df_ids)
-
-    train_df_ids['record_id_ferc'] = (
-        train_df_ids.Source + '_' +
-        train_df_ids.report_year.astype(str) + '_' +
-        train_df_ids.report_prd.astype(str) + '_' +
-        train_df_ids.respondent_id.astype(str) + '_' +
-        train_df_ids.spplmnt_num.astype(str)
-    )
-    if "row_number" in train_df_ids.columns:
-        train_df_ids["record_id_ferc"] = train_df_ids["record_id_ferc"] + \
-            "_" + train_df_ids.row_number.astype('Int64').astype(str)
+def prep_train_connections(file_path_training, plant_parts_df):
+    """
+    Get and prepare the training connections.
+
+    We have stored connections between ferc1 steam and the eia master unit
+    list. These records should be compiled of connections between these data
+    sets that are known to be correct because we will use these records to
+    train a machine learning model.
+
+    Args:
+        training_file_path (path-like): path to the CSV of training data. The
+            training data needs to have at least two columns: record_id_eia and
+            record_id_ferc1.
+        plant_parts_df (pandas.DataFrame): master unit list. generated from
+            `CompilePlantParts.generate_master_unit_list()` or
+            `get_master_unit_list_eia()`.
 
-    return train_df_ids.set_index(['record_id_ferc', 'record_id_eia'])
+    Returns:
+        pandas.DataFrame: training connections. A dataframe with has a
+        MultiIndex with record_id_eia and record_id_ferc1.
+    """
+    mul_cols = ['true_gran', 'appro_part_label',
+                'appro_record_id_eia', 'plant_part']
+    train_df = (
+        pd.read_csv(file_path_training,)
+        # we want to ensure that the records are associated with a
+        # "true granularity" - which is a way we filter out whether or not each
+        # record in the master unit list is actually a new/unique collection of
+        # plant parts
+        .merge(
+            plant_parts_df.reset_index()[['record_id_eia'] + mul_cols],
+            how='left', on=['record_id_eia'],
+        )
+        .assign(plant_part=lambda x: x['appro_part_label'],
+                record_id_eia=lambda x: x['appro_record_id_eia'])
+
+        # come light cleaning
+        .pipe(pudl.helpers.cleanstrings_snake, ['record_id_eia'])
```

Comment:
This line suggests that the EIA Record IDs in other contexts are case sensitive and contain whitespace. If so, is that desirable in those other contexts? Are there times when those differences are meaningful?

Reply:
comparing the ids means I need to care about case sensitivity. I think this problem was initially introduced from using RMI's training data which was in different cases. I've been generally ensuring these record ids are lowercase for easier comparisons. Perhaps I should just use `lower` instead of running this through `cleanstrings_snake`

Reply:
We also have a function called `strip_lower` which removes leading/trailing whitespace, compacts all interior whitespace to a single space, and lower-cases everything, which we've been using in other stringy comparisons.